frontend/store/auth-context.js

```diff
@@ -10,16 +11,32 @@ const AuthContext = React.createContext({
 
 export const AuthContextProvider = (props) => {
   const [token, setToken] = useState("");
-  const isLoggedIn = token.length !== 0;
+  const isLoggedIn = token.length !== 0
 
   const loginHandler = (token) => {
     setToken(token);
+    AsyncStorage.setItem('token',token);
   };
 
   const logoutHandler = () => {
     setToken("");
+    AsyncStorage.setItem('token',"");
   };
 
+  const isLogIn = async () => {
+      try {
+        let userToken = AsyncStorage.getItem('token');
+        console.log("before user token:", userToken);
+        setToken(await userToken);
```

Comment:
```suggestion
        let savedToken = await AsyncStorage.getItem('token');
        if (savedToken) setToken(await userToken);
```

Reply:
This should solve the problem that Ramy mentioned.

Reply:
The persistence isn't working.